tests/contrib/redis/test_redis_asyncio.py

```diff
@@ -0,0 +1,214 @@
+        assert val == "my-cheese"
+
+
+@pytest.mark.asyncio
+@pytest.mark.snapshot
+async def test_pin(redis_client):
+    Pin.override(redis_client, service="my-aioredis")
+    val = await redis_client.get("cheese")
+    assert val is None
+
+
+@pytest.mark.asyncio
+async def test_pipeline_traced(redis_client):
+    p = redis_client.pipeline(transaction=False)
+    p.set("blah", "boo")
+    p.set("foo", "bar")
+    p.get("blah")
+    p.get("foo")
+
+    response_list = await p.execute()
+    assert response_list[0] is True  # response from redis.set is OK if successfully pushed
+    assert response_list[1] is True
+    assert (
+        response_list[2].decode() == "boo"
+    )  # response from hset is 'Integer reply: The number of fields that were added.'
+    assert response_list[3].decode() == "bar"
+
+
+@pytest.mark.asyncio
+@pytest.mark.snapshot
+async def test_pipeline_traced_context_manager_transaction(redis_client):
+    """
+    Regression test for: https://github.com/DataDog/dd-trace-py/issues/3106
+
+    https://aioredis.readthedocs.io/en/latest/migration/#pipelines-and-transactions-multiexec
+
+    Example::
+
+        async def main():
+            redis = await aioredis.from_url("redis://localhost")
+            async with redis.pipeline(transaction=True) as pipe:
+                ok1, ok2 = await (pipe.set("key1", "value1").set("key2", "value2").execute())
+            assert ok1
+            assert ok2
+    """
+
+    async with redis_client.pipeline(transaction=True) as p:
+        set_1, set_2, get_1, get_2 = await (p.set("blah", "boo").set("foo", "bar").get("blah").get("foo").execute())
+
+    # response from redis.set is OK if successfully pushed
+    assert set_1 is True
+    assert set_2 is True
+    assert get_1.decode() == "boo"
+    assert get_2.decode() == "bar"
+
+
+@pytest.mark.asyncio
+async def test_two_traced_pipelines(redis_client):
+
+    with tracer.trace("web-request", service="test"):
+        if redis.VERSION >= (2, 0):
+            p1 = await redis_client.pipeline(transaction=False)
+            p2 = await redis_client.pipeline(transaction=False)
+            await p1.set("blah", "boo")
+            await p2.set("foo", "bar")
+            await p1.get("blah")
+            await p2.get("foo")
+        else:
+            p1 = redis_client.pipeline()
+            p2 = redis_client.pipeline()
+            p1.set("blah", "boo")
+            p2.set("foo", "bar")
+            p1.get("blah")
+            p2.get("foo")
+
+        response_list1 = await p1.execute()
+        response_list2 = await p2.execute()
+
+    assert response_list1[0] is True  # response from redis.set is OK if successfully pushed
+    assert response_list2[0] is True
+    assert (
+        response_list1[1].decode() == "boo"
+    )  # response from hset is 'Integer reply: The number of fields that were added.'
+    assert response_list2[1].decode() == "bar"
+
+
+@pytest.mark.asyncio
+async def test_parenting(redis_client):
```

Comment:
do you want to use the snapshot here too? Wondering since there's no assertion otherwise.

Reply:
Fixed.

Reply:
I'm not seeing any change?

Reply:
Fixed.